service/controller/chart/resource/namespace/create.go

```diff
@@ -58,3 +70,50 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 
 	return nil
 }
+
+func (r *Resource) ensureNamespaceUpdated(ctx context.Context, cr v1alpha1.Chart) error {
+	namespace, err := r.k8sClient.CoreV1().Namespaces().Get(ctx, key.Namespace(cr), metav1.GetOptions{})
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	updated := true
+
+	if namespace.GetLabels() == nil {
+		namespace.Labels = map[string]string{}
+	}
+
+	for k, v := range key.NamespaceLabels(cr) {
+		if namespace.GetLabels()[k] != v {
+			namespace.GetLabels()[k] = v
+			updated = false
+		}
+	}
+
+	if namespace.GetAnnotations() == nil {
+		namespace.Annotations = map[string]string{}
+	}
+
+	for k, v := range key.NamespaceAnnotations(cr) {
+		if namespace.GetAnnotations()[k] != v {
+			namespace.GetAnnotations()[k] = v
+			updated = false
+		}
+	}
+
+	if updated {
```

Comment:
Only if the operator saw there was an update on a map object, we update a namespace object.